core/listener.go

```diff
@@ -1,14 +1,60 @@
 package core
 
 import (
+	"context"
+	"net"
+
 	"github.com/quic-go/quic-go"
 )
 
 // A Listener for incoming connections
 type Listener interface {
-	quic.Listener
-	// Name listerner's name
-	Name() string
-	// Versions
-	Versions() []string
+	// Close the server. All active connections will be closed.
+	Close() error
+	// Addr returns the local network addr that the server is listening on.
+	Addr() net.Addr
+	// Accept returns new connections. It should be called in a loop.
+	Accept(context.Context) (Connection, error)
+}
+
+// A Connection is a connection between two peers.
+type Connection interface {
```

Comment:
Define a new `Connection` interface responsible for opening or accepting a stream.